hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3,18 +3,1146 @@
+            verifies the correctness of the reflected values once the chip goes into deep sleep.
+            This is replicated for DIO pins as well.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_pin_wake
+      desc: '''Verify pin wake up from deep sleep state.
+
+            Verify one of the 8 possible MIO or DIO pad inputs (randomly configured) can cause the
+            chip to wake up from sleep state. Verifying wake on posedge is sufficient for the chip
+            level integration testing. Upon wake up, SW reads the wake cause CSR to verify
+            correctness.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // PADCTRL tests:
+    {
+      name: chip_padctrl_attributes
+      desc: '''Verify pad attribute settings for all MIO and DIO pads.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // RV_DM (JTAG) tests:
+    {
+      name: chip_rv_dm_cpu_debug_mem
+      desc: '''Verify access to the debug mem from the CPU.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_rv_dm_jtag_debug_mem
+      desc: '''Verify access to the debug mem from the external JTAG interface.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_rv_dm_cpu_debug_req
+      desc: '''Verify debug request to Ibex while it is actively executing.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_rv_dm_ndm_reset_req
+      desc: '''Verify non-debug reset req initiated from RV_DM when the chip is awake.
+
+            Read CSRs / mem within all IPs in the chip to ensure that they are reset to the original
+            values. Read CSRs / mem in the debug domain to ensure that the values survive the reset.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_sleep_rv_dm_ndm_reset_req
+      desc: '''Verify non-debug reset req initiated from RV_DM when the chip is in deep sleep.
+
+            Read CSRs / mem within all IPs in the chip to ensure that they are reset to the original
+            values. Read CSRs / mem in the debug domain to ensure that the values survive the reset.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // PATTGEN (pre-verified IP) integration tests:
+    {
+      name: chip_pattgen_ios
+      desc: '''Verify pattern generation to chip output pads.
+
+            SW programs pattgen to generate distinct patterns on both groups. SW programs pinmux to
+            select pattgen outputs to be routed. SW validates the reception of patt_done interrupts.
+            Testbench verifies the correctness of the pattern seen on the IO pins.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // PWM (pre-verified IP) integration tests:
+    {
+      name: chip_sleep_pwm_ios_val
+      desc: '''Verify PWM signaling to chip output pads during deep sleep
```

Comment:
nice. Just as a sanity check, should we do both deep sleep and sleep? We do technically have multiple low power modes.

Reply:
Hmm, the benefit of checking normal sleep would be to ensure that the clk is not mis-connected. I think we should do an automated Jasper connectivity test to ensure all clocks and resets are connected properly.
 
With that, I don't see a lot of benefit of testing this in normal sleep. Plus deep sleep is already a super set of normal sleep. WDYT?